storybookjs/storybook
# Don't mangle function names for production builds

Issue:

Function names are mangled by UglifyJS during production build.
They are important for React stateless components, especially in a styleguide context such as Storybook. This allows playing nicely with third party libraries such as [`react-element-to-jsx-string`](https://github.com/algolia/react-element-to-jsx-string/blob/333591a19420cb52f93b7c638bcaa2ca10bf2d11/src/parser/parseReactElement.js#L12).
The issue happened after #2530.

## What I did

Passed the [`compress: { keep_fnames: true }`](https://github.com/mishoo/UglifyJS2/tree/harmony#compress-options) option to all production builds UglifyJS plugin.
Did not noticed significant bundle size increase on our internal styleguide, which is quite large.

## How to test

Is this testable with jest or storyshots? Nope

Does this need a new example in the kitchen sink apps? Nope

Does this need an update to the documentation? Nope

If your answer is yes to any of these, please make sure to include it in your PR.
